### ISSUE TITLE
Some endless running slugbuilder jobs and cluster backup runs for hours

### ISSUE THREAD

**Issue by philiplb** [flynn/flynn#2735]
Hi,

I just noticed that the cluster backup started at around 3am was still running (since 19h). I stopped it.
Then I noticed that there are some "zombie" jobs around always restarting if I flynn-host stop them:

> Flynn0-e93601f5-9939-4e1a-9c5b-741b00847d4b  starting                 myapp    slugbuilder
> Flynn0-ab3c9651-3303-4747-9dc0-d11317cd8f64  starting                 myapp    web
> Flynn0-54fae299-54de-4fb6-b432-9475a8f3a6bf  starting                 myapp    slugbuilder
> Flynn0-261d55c7-a9db-4699-87d0-52e3066494f2  starting                 myapp    web

This might be related to the endless backup.

I deleted the application but those jobs are just coming back as soon as I stop them.

Collecting debug info hangs on all three nodes at

> INFO[04-24|22:14:22] getting job logs

**Comment by titanous** [comment]
Please collect the logs in `/var/log/flynn` manually.

**Comment by philiplb** [comment]
Thank you, I've sent you an email.

**Comment by titanous** [comment]
@philiplb Have you checked the disk usage on the hosts? Also please post the `/var/log/upstart/flynn-host.log` files from each host.

**Comment by philiplb** [comment]
@titanous 
I mailed you the details. Disk usage seems ok on two of the three nodes. The first one doesn't respond to "df -h".

**Comment by titanous** [comment]
I think something is wrong with the host Flynn0, those jobs that are marked as starting are stuck. Especially combined with not being able to run `df`.

**Comment by philiplb** [comment]
Might a reboot be worth a try?

**Comment by titanous** [comment]
Yep, worth a try.

**Comment by philiplb** [comment]
No big luck but another round of logfiles.
Interesting: Every few hours, the status endpoint now delivers a 500, I think without content. The next request is always just fine. I added some flags to curl to maybe get more informations.

**Comment by philiplb** [comment]
Hm, just happened again and now it seems to be permanent. Getting a 500 without content.

```bash
curl -i -f -k -u :abc https://status.flynn.xyz.com
```

>  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
>                                 Dload  Upload   Total   Spent    Left  Speed
> 
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
>   0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
>   0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
> curl: (22) The requested URL returned error: 500 Internal Server Error

If you are interested, I would give you SSH access to the machines. Otherwise, I would restore from backup at some point the next day.

**Comment by lmars** [comment]
@philiplb drop the `-f` argument to `curl` and you should get some output.

**Comment by philiplb** [comment]
Ah, of course.
Here we go:
> {
>   "data": {
>     "status": "unhealthy",
>     "detail": {
>       "blobstore": {
>         "status": "unhealthy"
>       },
>       "controller": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "controller-scheduler": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "controller-worker": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "dashboard": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "discoverd": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "flannel": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "gitreceive": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "logaggregator": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       },
>       "mariadb": {
>         "status": "unhealthy"
>       },
>       "postgres": {
>         "status": "unhealthy"
>       },
>       "router": {
>         "status": "healthy",
>         "version": "v20160327.0"
>       }
>     },
>     "version": "v20160327.0"
>   }
> }

**Comment by philiplb** [comment]
Maybe this is related? From the controller DB:

> SELECT * FROM que_jobs;
>  priority |            run_at             | job_id |  job_class   |                                                                                                                            args                                                                                                                             | error_count |           last_error           | queue |         locked_until
> ----------+-------------------------------+--------+--------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+-------------+--------------------------------+-------+-------------------------------
>       100 | 2016-04-27 09:40:27.30021+00  |     32 | app_deletion | {"id":"5201c909-c475-4be1-b63c-3cb5286c4de2","name":"eltern","meta":{},"strategy":"all-at-once","release":"8dc0761f-1409-44fd-ad0d-3f1cba07ca7c","deploy_timeout":120,"created_at":"2016-04-21T21:47:05.551958Z","updated_at":"2016-04-24T13:54:23.93602Z"} |          16 | controller: resource not found |       | 2016-04-26 15:28:17.698012+00
>       100 | 2016-04-27 09:42:29.040877+00 |     33 | app_deletion | {"id":"5201c909-c475-4be1-b63c-3cb5286c4de2","name":"eltern","meta":{},"strategy":"all-at-once","release":"8dc0761f-1409-44fd-ad0d-3f1cba07ca7c","deploy_timeout":120,"created_at":"2016-04-21T21:47:05.551958Z","updated_at":"2016-04-24T13:54:23.93602Z"} |          16 | controller: resource not found |       | 2016-04-26 15:30:18.541798+00
> (2 rows)

Would it be a way to recover to delete the endless starting jobs from the Controller-Postgres? If so, what tables and how would I delete them?

**Comment by titanous** [comment]
@philiplb I believe this was fixed in #2747.

**Comment by philiplb** [comment]
Ou, awesome. When does this hit the stable channel for an update? :)

**Comment by titanous** [comment]
@philiplb Likely tomorrow.

**Comment by titanous** [comment]
@philiplb A nightly release is out that contains the fix for this, it will be promoted to stable soon. Feel free to try it out!

**Comment by philiplb** [comment]
Will do ASAP, thanks. :)

Small status update: The status endpoint is behaving nicely now the last days after I killed some Postgres- and MySQL-Jobs so they restarted again.

The backup stopped to run endlessly then and failed with:

> mysqldump: Got error: 2003: "Can't connect to MySQL server on 'leader.mariadb.discoverd' (111 "Connection refused")" when trying to connect
> error dumping mariadb database: error running mariadb export: unexpected command exit status 2

So I killed the MySQL again. This didn't fix this problem. So I scaled it down to 0 as I don't use it (yet) anyway.

Now the backup is working again. :) With a whooping size of 1.9GB without that many releases. Will investigate at some point after switching to the new stable version.
